### PR TITLE
Feature/full event view 2

### PR DESCRIPTION
**Pull request by SRozhina**
Implement full event view

### CODE REVIEW

File: ITEvents/Screens/Event view/EventPlace.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import Foundation
+import UIKit
+
+class EventPlace: DecorationView {
+    @IBOutlet weak var cityLabel: UILabel!
+    @IBOutlet weak var addressLabel: UILabel!
+    
+    class func initiateAndSetup(withCity city: String, country: String, address: String) -> EventPlace {
+        let eventPlaceView = UINib(nibName: "EventPlace", bundle: nil).instantiate(withOwner: nil, options: nil)[0] as! EventPlace
+        //eventPlaceView.cityLabel.text = "\(city), \(country)"
```
Comment (AVykhrystyuk) [inline ITEvents/Screens/Event view/EventPlace.swift]: do you need the comments here?

File: ITEvents/Screens/Event view/EventInfo.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import Foundation
+import UIKit
+
+class EventInfo: DecorationView {
+    @IBOutlet weak var imageView: RoundedImageView!
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var dateLabel: UILabel!
+    
+    class func initiateAndSetup(withImage image: UIImage, title: String, date: String) -> EventInfo {
+        let eventInfoView = UINib(nibName: "EventInfo", bundle: nil).instantiate(withOwner: nil, options: nil)[0] as! EventInfo
```
Comment (AVykhrystyuk) [inline ITEvents/Screens/Event view/EventInfo.swift]:
```
let eventInfoView = ViewSharedUtils.createNib<EventInfoVIew>(name: "EventInfo")
...
class ViewSharedUtils {
    createNib<T>(name: string) -> T { 
        return UINib(nibName: name, bundle: nil).instantiate(withOwner: nil, options: nil)[0] as! T
   }
}
```

File: ITEvents/Screens/Event view/EventPlace.swift [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import Foundation
+import UIKit
+
+class EventPlace: DecorationView {
+    @IBOutlet weak var cityLabel: UILabel!
+    @IBOutlet weak var addressLabel: UILabel!
+    
+    class func initiateAndSetup(withCity city: String, country: String, address: String) -> EventPlace {
+        let eventPlaceView = UINib(nibName: "EventPlace", bundle: nil).instantiate(withOwner: nil, options: nil)[0] as! EventPlace
```
Comment (AVykhrystyuk) [inline ITEvents/Screens/Event view/EventPlace.swift]: `let eventPlaceView = ViewSharedUtils.createNib<EventPlaceVIew>(name: "EventPlace")`

File: ITEvents/Screens/Event view/EventInfo.swift [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import Foundation
+import UIKit
+
+class EventInfo: DecorationView {
```
Comment (AVykhrystyuk) [inline ITEvents/Screens/Event view/EventInfo.swift]: EventInfoView: UIView

File: ITEvents/Screens/Event view/EventTags.swift [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import Foundation
+import UIKit
+
+class EventTags: DecorationView {
```
Comment (AVykhrystyuk) [inline ITEvents/Screens/Event view/EventTags.swift]: same here

File: ITEvents/Screens/Event view/EventView.swift [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import Foundation
+import UIKit
+
+class EventView: UIStackView {
+    
+    class func initiateAndSetup(with event: Event) -> EventView {
+        let tempDescription = "19:10 - 19:40: “Svelte — «магический» фреймворк” Тим Маринин Поговорим про Svelte и подумаем зачем нам ещё один фреймворк, и что это значит, что в нём нет рантайма? 19:55 - 20:25: “ES2018?” Михаил Полубояринов 23–28 января 2018 г. прошел очередной митинг TC39, на котором обсуждалось то что пойдет в ES2018 и Михаилу хотелось бы поделиться с Вами результатами этого митинга. 20:40 - 21:20: “рефакторинг” Алексей Золотых Рефакторинг приложения является неотъемлемой частью разработки. В своем докладе я расскажу про практические аспекты этого процесса."
+        
+        let eventView = UINib(nibName: "EventView", bundle: nil).instantiate(withOwner: nil, options: nil)[0] as! EventView
```
Comment (AVykhrystyuk) [inline ITEvents/Screens/Event view/EventView.swift]: `let eventView = ViewSharedUtils.createNib<EventView>(name: "EventView")`

File: ITEvents/Screens/Event view/RoundedImageView.swift [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import Foundation
+import UIKit
+
+class RoundedImageView: UIImageView {
```
Comment (AVykhrystyuk) [inline ITEvents/Screens/Event view/RoundedImageView.swift]: need for a common setup here
```
setup() {
     layer.cornerRadius = 14
     clipsToBounds = true
}
```

File: ITEvents/Screens/Event view/ShowEventSource.swift [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import Foundation
+import UIKit
+
+class ShowEventSource: DecorationView {
+    @IBOutlet weak var showSourceButton: UIButton!
+    
+    class func initiateAndSetup(with source: String) -> ShowEventSource {
+        let showEventSourceView = UINib(nibName: "ShowEventSource", bundle: nil).instantiate(withOwner: nil, options: nil)[0] as! ShowEventSource
```
Comment (AVykhrystyuk) [inline ITEvents/Screens/Event view/ShowEventSource.swift]: don't forget to update

File: ITEvents/Screens/Event view/ShowMoreDescriptions.swift [hunk 9]
```diff
@@ -0,0 +1,22 @@
+import Foundation
+import UIKit
+
+class ShowMoreDescriptions: DecorationView {
+    @IBOutlet weak var showMoreDescriptionsButton: UIButton!
+    
+    //TODO create protocol for subviews??
+    class func initiateAndSetup(with descriptionsCount: Int) -> ShowMoreDescriptions {
+        let showMoreDescriptionsView = UINib(nibName: "ShowMoreDescriptions",bundle: nil).instantiate(withOwner: nil, options: nil)[0] as! ShowMoreDescriptions
```
Comment (AVykhrystyuk) [inline ITEvents/Screens/Event view/ShowMoreDescriptions.swift]: don't forget to update

File: ITEvents/Screens/Event view/ShowMoreDescriptions.swift [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import Foundation
+import UIKit
+
+class ShowMoreDescriptions: DecorationView {
```
Comment (AVykhrystyuk) [inline ITEvents/Screens/Event view/ShowMoreDescriptions.swift]: ShowMoreEventsVIew

File: ITEvents/Screens/Event view/ShowMoreDescriptions.swift [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+    class func initiateAndSetup(with descriptionsCount: Int) -> ShowMoreDescriptions {
+        let showMoreDescriptionsView = UINib(nibName: "ShowMoreDescriptions",bundle: nil).instantiate(withOwner: nil, options: nil)[0] as! ShowMoreDescriptions
+        let buttonTitle = ShowMoreDescriptions.getDescriptionFor(hidden: true, count: descriptionsCount)
+        showMoreDescriptionsView.showMoreDescriptionsButton.setTitle(buttonTitle, for: .normal)
+        return showMoreDescriptionsView
+    }
+    
+    class func getDescriptionFor(hidden: Bool, count: Int) -> String {
+        if hidden {
+            return count == 1 ? "Show 1 more description" : "Show \(count) more descriptions"
```
Comment (AVykhrystyuk) [inline ITEvents/Screens/Event view/ShowMoreDescriptions.swift]:
```
return count == 1 
            ? "Show one more description" 
            : "Show \(count) more descriptions"
```

File: ITEvents/Screens/Event view/ShowMoreDescriptions.swift [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+        } else {
```
Comment (AVykhrystyuk) [inline ITEvents/Screens/Event view/ShowMoreDescriptions.swift]: else can be omitted here

File: ITEvents/Screens/FullEventView/FullEventViewController.swift [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+        showMoreDescriptions.showMoreDescriptionsButton.addTarget(self, action: #selector(expandOrCollapseDescriptions), for: .touchUpInside)
+        stackView.addArrangedSubview(showMoreDescriptions)
+        
+        let descriptionsStackView = UIStackView()
+        descriptionsStackView.axis = .vertical
+        descriptionsStackView.spacing = 15
+        descriptionsStackView.tag = 123
+        for _ in 0..<descriptionsCount {
+            let eventView = EventView.initiateAndSetup(with: event)
+            let sourceLabel = createLabelFor(source: "Event from Meetabit.com")
```
Comment (AVykhrystyuk) [inline ITEvents/Screens/FullEventView/FullEventViewController.swift]:
```
createSourceLabel(text: "Event from Meetabit.com")
```
or
```
func createLabelFor(source text: String) -> UILabel {
```

File: ITEvents/Screens/FullEventView/FullEventViewController.swift [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+        
+        let descriptionsCount = 2
+        let showMoreDescriptions = ShowMoreDescriptions.initiateAndSetup(with: descriptionsCount)
+        showMoreDescriptions.showMoreDescriptionsButton.addTarget(self, action: #selector(expandOrCollapseDescriptions), for: .touchUpInside)
+        stackView.addArrangedSubview(showMoreDescriptions)
+        
+        let descriptionsStackView = UIStackView()
+        descriptionsStackView.axis = .vertical
+        descriptionsStackView.spacing = 15
+        descriptionsStackView.tag = 123
```
Comment (AVykhrystyuk) [inline ITEvents/Screens/FullEventView/FullEventViewController.swift]: stop using tag
start using `private let descriptionsStackView: UIStackView`

File: ITEvents/Screens/Event view/EventPlace.swift (hunk 1 above)
Comment (SRozhina) [reply ITEvents/Screens/Event view/EventPlace.swift]: Yes, they needed for previous realization

File: ITEvents/Screens/Event view/RoundedImageView.swift (hunk 7 above)
Comment (SRozhina) [reply ITEvents/Screens/Event view/RoundedImageView.swift]: made

**Pull request merged**